src/attributes/codegen.md

```diff
@@ -140,6 +140,28 @@ feature detection on the x86 platforms.
 > may be enabled or disabled for an entire crate with the
 > [`-C target-feature`] flag.
 
+## The `track_caller` attribute
+
+The `track_caller` attribute applies to function declarations, allowing code within the function to
+observe the callsite of the topmost `track_caller`-attributed function in the attributed function's
+callers. If the attributed function itself is called by an unattributed function, then the
+attributed function observes its own callsite. If the attributed function is called by another
+attributed function, then code within the callee observes the callsite of the caller, and so on.
+
+> Note: `rustc` implements the `core::intrinsics::caller_location` intrinsic for observing the
+> callsite of a function with `#[track_caller]`, wrapped by `core::panic::Location::caller` for
+> general use.
+
+Coercing a function with `#[track_caller]` to a function pointer creates a shim which appears to
+observers to have been called at the attributed function's definition site.
+
+> Note: Rules around shims for function pointers are necessary because `rustc` implements
```

Comment:
Just to be clear, by "rules" you mean the coercion above? If so, can you reword to "The aforementioned shim for function pointers..."?